packages/module/src/hooks/filtering/useClientFilterDerivedState.ts

```diff
@@ -15,25 +15,23 @@ export interface UseClientFilterDerivedStateArgs<TItem, TFilterCategoryKey exten
    */
   items: TItem[];
   /**
-   * Definitions of the filters to be used (must include `getItemValue` functions for each category when performing filtering locally)
+   * Feature-specific args: A subset of the `TableState` object's `filter` property with the state itself and relevant state args
    */
-  filterCategories?: FilterCategory<TItem, TFilterCategoryKey>[];
-  /**
-   * The "source of truth" state for the filter feature (returned by useFilterState)
-   */
-  filterState: FilterState<TFilterCategoryKey>;
+  filter: FilterState<TFilterCategoryKey> & Pick<FilterStateArgs<TItem, TFilterCategoryKey>, 'filterCategories'>;
 }
 
 /**
  * Given the "source of truth" state for the filter feature and additional arguments, returns "derived state" values and convenience functions.
  * - For local/client-computed tables only. Performs the actual filtering logic, which is done on the server for server-computed tables.
  * - "source of truth" (persisted) state and "derived state" are kept separate to prevent out-of-sync duplicated state.
  */
-export const useClientFilterDerivedState = <TItem, TFilterCategoryKey extends string>({
-  items,
-  filterCategories = [],
-  filterState: { filterValues }
-}: UseClientFilterDerivedStateArgs<TItem, TFilterCategoryKey>) => {
+export const useClientFilterDerivedState = <TItem, TFilterCategoryKey extends string>(
```

Comment:
The `TFilterCategoryKey` must extend string because there are some places where string operations are performed on it (I think?), so that `extends` is telling TS that it is a union of specific strings but we can do string stuff with it. This is also the case with some of the other generics. They could all probably use another look eventually, especially in places where I make them optional and default to `string` which is probably not actually desired.